runway/data_types.py

```diff
@@ -109,14 +115,14 @@ def setup(opts):
     :type max_length: int, optional
     :raises MissingArgumentError: A missing argument error if item_type is not specified
     """
-    def __init__(self, item_type=None, name=None, description=None, min_length=0, max_length=None):
-        super(array, self).__init__('array', name=name, description=description)
+    def __init__(self, item_type=None, description=None, min_length=0, max_length=None):
+        super(array, self).__init__('array', description=description)
         if item_type is None: raise MissingArgumentError('item_type')
         if inspect.isclass(item_type):
             self.item_type = item_type()
         else:
             self.item_type = item_type
-        self.name = name or '%s_array' % self.item_type.name
+        self.item_type.name = '%s_array_item' % self.item_type.type
         self.min_length = min_length
```

Comment:
Notice that we are no longer assigning the name of the `array` data type itself during construction, but are instead assigning a default name to the `item_type` instead. Right now there is no way for a user/model porter to name the variables in an array. That should be fine, but its worth noting. Without this line they would all have the name `None` (which is perhaps preferable?).